boefjes/boefjes/plugins/kat_maxmind_geoip/main.py

```diff
@@ -0,0 +1,121 @@
+import hashlib
+import io
+import json
+import os
+import re
+import shutil
+import tarfile
+from datetime import datetime, timezone
+from os import getenv
+from pathlib import Path
+
+import maxminddb
+import requests
+
+from boefjes.job_models import BoefjeMeta
+
+BASE_PATH = Path(getenv("OPENKAT_CACHE_PATH", Path(__file__).parent))
+GEOIP_PATH_PATTERN = r"GeoLite2-City_\d+/GeoLite2-City.mmdb"
+GEOIP_META_PATH = BASE_PATH / "geoip-meta.json"
+GEOIP_SOURCE_URL = "https://download.maxmind.com/geoip/databases/GeoLite2-City/download?suffix=tar.gz"
+GEOIP_CACHE_TIMEOUT = 7200  # in seconds
+HASHFUNC = "sha256"
+REQUEST_TIMEOUT = 30
+
+
+def run(boefje_meta: BoefjeMeta) -> list[tuple[set, bytes | str]]:
+    input_ = boefje_meta.arguments["input"]
+    hash_algorithm = getenv("HASHFUNC", HASHFUNC)
+
+    if not geoip_file_exists() or cache_out_of_date():
+        geoip_meta = refresh_geoip(hash_algorithm)
+    else:
+        with GEOIP_META_PATH.open() as json_meta_file:
+            geoip_meta = json.load(json_meta_file)
+
+    geoip_path = find_geoip_path()
+
+    with maxminddb.open_database(geoip_path) as reader:
+        results = reader.get(input_["address"])
+
+    return [
+        ({"maxmind-geoip/geo_data"}, json.dumps(results)),
+        (
+            {"maxmind-geoip/cache-meta"},
+            json.dumps(geoip_meta),
+        ),
+    ]
+
+
+def create_hash(data: bytes, algo: str) -> str:
+    hashfunc = getattr(hashlib, algo)
+    return hashfunc(data).hexdigest()
+
+
+def cache_out_of_date() -> bool:
+    """Returns True if the file is older than the allowed cache_timout"""
+    now = datetime.now(timezone.utc)
+    max_age = int(getenv("GEOIP_CACHE_TIMEOUT", GEOIP_CACHE_TIMEOUT))
```

Comment:
Im missing this in the jsonschema

Reply:
Created a new ticket for this, but ill fix in this PR. Wanted to keep momentum initially.

https://github.com/minvws/nl-kat-coordination/issues/3301